packages/blockchain-wallet-v4-frontend/src/scenes/DebitCard/CardDashboard/ManageCardBox/index.tsx

```diff
@@ -24,10 +25,25 @@ const SwitchWrapper = styled.div`
   margin: auto;
 `
 
-const ManageCardBox = ({ cards, debitCardActions, lockHandler, modalActions }) => {
+const ManageCardBox = ({ alertActions, cards, debitCardActions, lockHandler, modalActions }) => {
   const activeCard = cards[0]
   const [isCardLocked, setCardLocked] = useState(activeCard.status === CardStateType.LOCKED)
 
+  useEffect(() => {
+    lockHandler.cata({
```

Comment:
I think this could end up having weird side effects. Try using the new `useRemote()` hook that @awiermanncasas-bc created. Outside of the useEffect you would have something 
```
 const { hasData, hasError } = useRemote(lockHandler)

useEffect(() => {
  if (hasData && !hasError) {
    alertActions.displaySuccess(C.DEBIT_CARD_DASHBOARD_LOCK_SUCCESS)
  } else if (!hasData && hasError) {
    setCardLocked(!isCardLocked)
    alertActions.displayError(C.DEBIT_CARD_DASHBOARD_LOCK_ERROR)
  }
}, [hasData, hasError])
```

Reply:
I was thinking particularly about if `lockHandler` were to change too often it would cause the useEffect to rerender the whole component, but after looking at this again, even in my example if lockHandler were to change the component would rerender so maybe they are functionally the same. Stylistically though with useRemote, you don't have to include the loading and NotAsked states since they don't do anything anyways. I guess it's up to you to decide which way you like best.

Reply:
I prefer this way because I think it is easier to read and understand what it does in each state. In the other case you mentioned, you have to read 2 different if clauses and try to understand which state of the request it is handleling